src/shared/dissect-image.c

```diff
@@ -4008,9 +4010,35 @@ int verity_dissect_and_mount(
 
                 assert(!isempty(required_host_os_release_id));
 
-                r = load_extension_release_pairs(dest, IMAGE_SYSEXT, dissected_image->image_name, relax_extension_release_check, &extension_release);
+                if (dest)
+                        r = load_extension_release_pairs(
+                                        dest,
+                                        IMAGE_SYSEXT,
+                                        dissected_image->image_name,
+                                        relax_extension_release_check,
+                                        &extension_release);
+                else
+                        r = load_extension_release_pairs_at(
+                                        dissected_image->partitions[PARTITION_ROOT].fsmount_fd,
```

Comment:
this doesn't look right, because the file might be us the USR tree

Reply:
we treat all extensions as 'root' partitions, ie they have to include the full hierarchy

Reply:
we do? where? i mean a confext has to, but a sysext should probably always include just a usr partition, no root partition

Reply:
Everywhere. A sysext is expected to include /usr/... and/or /opt/... it cannot just contain lib/systemd/...

Reply:
hmm? 

So you are saying a sysext DDI with just a PARTITION_USR partition is fine then? That's what I was saying. but your code only looks in PARTITION_ROOT

Reply:
If I understand correctly we need /usr in the image, so that means PARTITION_USR doesn't work, because it doesn't have /usr. /usr is / when the partition is PARTITION_USR.

Reply:
i don't follow? if you mount a DDI with only PARTITION_USR then the dissector will automatically mount a tmpfs to the root, and the user partition under /usr below it.

Reply:
No, that does not work, and it never worked for extensions, they have to be 'root' partitions. And it makes sense - they can also ship a /opt hierarchy

Reply:
"can" doesn't mean "must"...

sorry, but this really should work. For 90% of all sysext extensions it makes sense to only do the /usr/ thing, and forget about the rest.

Reply:
Once again: it does not work, not just here, but everywhere else too, it's just not supported, at all, anywhere

Reply:
DDIs without root fs are supported fine generally, and they should also be supported for sysext if they aren't right now. (but I would assume they just work)

hence, please do this right, we *should* support them

Reply:
The tunnel stuff is not exclusive to extensions so it's not using this code anyway, I'll remove these changes as they are not needed